src/gui/mainwindow.cpp

```diff
@@ -1469,50 +1478,33 @@ void MainWindow::showStatusBar(bool show)
     }
 }
 
-void MainWindow::loadPreferences(const bool configureSession)
+void MainWindow::loadPreferences()
 {
-    const Preferences *const pref = Preferences::instance();
-#ifdef Q_OS_MACOS
-    Q_UNUSED(configureSession);
-#else
-    const bool newSystrayIntegration = pref->systrayIntegration();
-    m_ui->actionLock->setVisible(newSystrayIntegration);
-    if (newSystrayIntegration != (m_systrayIcon != nullptr))
+    const Preferences *pref = Preferences::instance();
+
+#ifndef Q_OS_MACOS
+    // system tray icon
+    const bool useSystray = pref->systemTrayEnabled();
+    m_ui->actionLock->setVisible(useSystray);
```

Comment:
Shouldn't it become visible only after the tray icon is really created?

Reply:
Added another commit for it.